.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseHolidayParser.cs

```diff
@@ -28,6 +30,24 @@ public ParseResult Parse(ExtractResult result)
         }
 
         public DateTimeParseResult Parse(ExtractResult er, DateObject refDate)
+        {
+            if (string.IsNullOrEmpty(er.Metadata?.HolidayName))

```

Comment:
This check should be on the weekend. Not on the holiday name itself, right? Perhaps have two separate pieces of metadata?

Reply:
Definitely agreed. For readability, I have added two pieces of metadata: 
- metadata.IsHolidayRange
- metadata.IsHolidayWeekend
This would also allow us to later extend holiday ranges for other non-weekend holidays without having to revisit the logic of the flow.